test/fishjam/cluster/api_test.exs

```diff
@@ -0,0 +1,248 @@
+defmodule Fishjam.Cluster.ApiTest do
+  @moduledoc false
+
+  # These tests can only be run with `mix test.cluster.epmd` or `mix test.cluster.dns`.
+
+  use ExUnit.Case, async: false
+
+  @token Application.compile_env(:fishjam, :server_api_token)
+  @headers [Authorization: "Bearer #{@token}", Accept: "Application/json; Charset=utf-8"]
+  @post_headers @headers ++ [{"Content-Type", "application/json"}]
+  @nodes ["app1:4001", "app2:4002"]
+
+  @moduletag :cluster
+  @max_test_duration 400_000
+
+  setup do
+    if Mix.env() != :test_cluster do
+      raise "Load balancing tests can only be run with MIX_ENV=test_cluster"
+    end
+
+    # Delete all leftover rooms after each test
+    on_exit(fn ->
+      for node <- @nodes do
+        with {:ok, %HTTPoison.Response{status_code: 200, body: body}} <-
+               HTTPoison.get("http://#{node}/test/local/room", @headers) do
+          rooms = body |> Jason.decode!() |> Map.get("data")
+
+          for room <- rooms, do: HTTPoison.delete("http://#{node}/room/#{room["id"]}", @headers)
+        end
+      end
+    end)
+
+    {:ok, %{nodes: @nodes}}
+  end
+
+  @tag timeout: @max_test_duration
+  test "load-balancing and listing rooms in a cluster", %{nodes: [node1, node2]} do
+    response_body1 = add_room(node1)
+
+    fishjam_instance1 = get_fishjam_address(response_body1)
+
+    assert_room_count_on_fishjam(fishjam_instance1, 1)
+
+    assert_room_count_in_cluster(node1, 1)
+    assert_room_count_in_cluster(node2, 1)
+
+    response_body2 = add_room(node1)
+
+    fishjam_instance2 = get_fishjam_address(response_body2)
+
+    assert_room_count_on_fishjam(fishjam_instance2, 1)
+
+    assert_room_count_on_fishjam(node1, 1)
+    assert_room_count_on_fishjam(node2, 1)
+
+    assert_room_count_in_cluster(node1, 2)
+    assert_room_count_in_cluster(node2, 2)
+
+    room_id = response_body1 |> Jason.decode!() |> get_in(["data", "room", "id"])
+
+    delete_room(fishjam_instance1, room_id)
+
+    assert_room_count_on_fishjam(fishjam_instance1, 0)
+    assert_room_count_on_fishjam(fishjam_instance2, 1)
+
+    assert_room_count_in_cluster(node1, 1)
+    assert_room_count_in_cluster(node2, 1)
```

Comment:
I think this could be moved to a helper function which would be provided with a number of rooms on each node and it could verify that each node has a correct number of rooms and whole cluster API also.